logistics_project/apps/malawi/handlers/receipts.py

```diff
@@ -53,7 +53,7 @@ def handle(self, text):
         # parse the report and save as normal receipt
         stock_report = ProductReportsHelper(self.msg.logistics_contact.supply_point,
                                             Reports.REC, self.msg.logger_msg)
-        stock_report.parse(text)
+        stock_report.newparse(text)
```

Comment:
From what i could tell from looking at the code and testing, `newparse` is the same as `parse` for the malawi format and populates the same information on the stock report. `newparse` is used everywhere else for malawi though so it makes it easier to integrate with things like error handling. By making this switch, the receipts handler will now gracefully handle invalid product codes.